assignment7.txt

```diff
@@ -0,0 +1,14 @@
+Lloyd Page
+1. ++*p dereferences the value that p points to and increments it. *p++ increments the pointer, and then dereferences it. *++p increments the pointer and then dereferences it.
+2. No, the operands on the same level as prefix ++ and = read right to left,all other operands read left to right.
+3. Dynamic starting size arrays, pass by reference for values, Dynamic memory allocation, provide a way to modify fuction calling arguments, more efficient use of memory and time in regards to arrays, multiple return functions, return arrays
+4.1 char *, its a string, which is a char[] with a '\0', and all arrays are pointers of their respective data type
+4.2 Invalid, xyz is not a declared array
```

Comment:
Since it's in quotations, it's automatically an array of characters (by definition that is what a string is). So the "xyz"[1] is just accessing "y". Then you just subtract 'y' from it to get 0. This is because, if you can recall, a char is just a value mapped to a character!